File: quizServer/src/main/java/com/exam/service/QuizService.java
package com.exam.service;

import java.util.List;
import java.util.Set;

import org.springframework.http.ResponseEntity;

import com.exam.model.exam.Category;
import com.exam.model.exam.Question;
import com.exam.model.exam.Quiz;


public interface QuizService {

	public Quiz addQuiz(Quiz quiz);
	public Quiz updateQuiz(Quiz quiz);
	public Set<Quiz> getQuizzes();
	public Quiz getQuiz(long quizId);
	public boolean deleteQuiz(Long quizId);
	List<Quiz> getQuizzesOfCategory(Category category);
	List<Quiz> getQuizzesOfCategoryAndActive(Category category,Boolean isActive);
	
}

File: quizFront/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
   providedIn: 'root'
})
export class QuizService {

   constructor(private http: HttpClient) { }

   public quizzes() {
      return this.http.get(`${baseUrl}/quiz/`,);
   }

   public addQuiz(quiz: any) {
      return this.http.post(`${baseUrl}/quiz/`, quiz);
   }

   public deleteQuiz(qId: any) {
      return this.http.delete(`${baseUrl}/quiz/${qId}`);
   }

   public getQuiz(qId: any) {
      return this.http.get(`${baseUrl}/question/quiz/${qId}`);
   }

   public getQuizById(qId: any) {
      return this.http.get(`${baseUrl}/quiz/${qId}`);
   }

   public updateQuiz(quiz: any) {
      return this.http.put(`${baseUrl}/quiz/`, quiz);
   }

   public getQuizzesOfCategory(cid: any) {
      return this.http.get(`${baseUrl}/quiz/category/${cid}`);
   }

   public getQuizzesOfCategoryAndActive(cid: any, isActive: any) {
      return this.http.get(`${baseUrl}/quiz/category/${cid}/${isActive}`);
   }

}
